Migrate wasm languages to dev docs (#1177)
fermyon/developer: content/wasm-languages/index.md

tags = ["webassembly", "programming languages"]
template = "page"
[extra]
author = "Fermyon Staff"
url = "https://github.com/fermyon/developer/blob/main/content/wasm-languages/index.md"

---

A wide variety of languages can be compiled to WebAssembly. Our programming language guide focuses on the top 20 programming languages. It also covers interesting programming languages as well as programming languages written specifically for WebAssembly.

The [Language Matrix](/wasm-languages/webassembly-language-support) provides an easy reference for which languages support what features of WebAssembly. It also links to detailed pages for each programming language that Fermyon tracks.

Fermyon WebAssembly Languages does not track WebAssembly execution environments or runtimes. For example, we track ways to compile [PHP](/wasm-languages/php) to WebAssembly, but we do not track how to execute WebAssembly in a PHP program.

## Help Us Stay Current

We do our best to keep the information here up to date. But with so many languages, each evolving rapidly, we're bound to miss a thing or two. If you have a correction or update, ping us on Twitter [@fermyontech](https://twitter.com/fermyontech).

> >> If you are interested in contributing to this guide, head on over to [the GitHub repo](https://github.com/fermyon/wasm-languages).

Author: ```suggestion
>> If you are interested in contributing to this guide, head on over to [the GitHub repo](https://github.com/fermyon/developer/wasm-languages).
```

## Adding Your Language to the Matrix

> If you are affiliated with a language project that adds WebAssembly support to a language, ping us on Twitter [@fermyontech](https://twitter.com/fermyontech) and let us know. We are particularly interested in WebAssembly-specific languages as well as WASI support for top-20 languages. When assessing new projects, we do prefer stable and complete codebases over experimental ones. However, given how young the WebAssembly world is, this is not a requirement.

Author: ```suggestion
If you are affiliated with a language project that adds WebAssembly support to a language, please consider [contributing](./CONTRIBUTING.md). We are particularly interested in WebAssembly-specific languages and WASI support for top-20 languages. When assessing new projects, we prefer stable and complete codebases over experimental ones. However, given how young the WebAssembly world is, this is not a requirement.
```
